pkg/config/config_template.yaml

```diff
@@ -413,6 +413,125 @@ api_key:
 #
 # secret_backend_timeout: 5
 
+## @param snmp_listener - custom object - optional
+## Creates and schedules an snmp listener to automatically discover your SNMP devices.
+## Discover devices can then be monitored with the snmp integration by using
+## the provided auto_conf.yaml file.
+#
+# snmp_listener:
+
+  ## @param workers - integer - optional - default: 2
+  ## Number of goroutines used to discover SNMP devices. Increasing this value
+  ## will help to discover devices faster but at the cost of increased resource consumption.
+  #
+  # workers: 2
+
+  ## @param discovery_interval - integer - optional - default: 3600
+  ## How often to discover new SNMP devices. Decreasing this value
+  ## will help to discover devices faster but at the cost of increased resource consumption.
+  #
+  # discovery_interval: 3600
+
+  ## @param allowed_failures - integer - optional - default: 3
+  ## Number of failed requests to a given SNMP device before removing it for the list of monitored
+  ## devices.
+  ## If a device shuts down, the Agent will stop monitoring it after `discovery_interval * allowed_failures`
+  #
+  # allowed_failures: 3
```

Comment:
@therve QQ, any reason we have those options are at global level instead of being available for each per network config ?

(not related to this PR per se)

Reply:
For the interval, because we reserve the option to mix subnets when we discovery (right now it's sequential, but it changed during the PR writing). So it's meant to be "discovery step, pause".

For the allowed failures, mostly because I don't think it's worth having it more granular.

Reply:
What do you mean by `to mix subnets`?

Reply:
I mean handle one ip from one subnet and then from another, instead of doing all the ips from one subnet as we do now.